demos/refactory/index.js

```diff
@@ -0,0 +1,40 @@
+import _ from 'underscore';
+import $ from 'jquery';
+import Backbone from 'backbone';
+import 'bootstrap-webpack';
+import TableView from '../../js/table-view.js'
+import ProjectionChain from '../../js/projection-chain/pg-chain.js'
+import odata from '../../js/projection-chain/odata.js'
+
+function renderTable (state, {} = {}) {
+	var promiseData = state.then(function (data) {
```

Comment:
By convention, we use `p$` prefix for promises. However, it's not necessary for you to define a temporary variable here. You can return it directly.